app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -237,15 +237,15 @@ public void onPageStop(GeckoSession session, boolean success) {
 
                 @Override
                 public void onSecurityChange(GeckoSession session,
-                                             GeckoSession.ProgressListener.SecurityInformation securityInfo) {
+                                             GeckoSession.ProgressDelegate.SecurityInformation securityInfo) {
                     // TODO: Split current onPageFinished() callback into two: page finished + security changed
                     isSecure = securityInfo.isSecure;
                 }
             };
         }
 
-        private GeckoSession.NavigationListener createNavigationListener() {
-            return new GeckoSession.NavigationListener() {
+        private GeckoSession.NavigationDelegate createNavigationDelegate() {
```

Comment:
\[PMD\] INFO: The method 'createNavigationDelegate' has a Cyclomatic Complexity of 9\.